libp2p/stream_muxer/yamux/yamux.py

```diff
@@ -0,0 +1,270 @@
+"""
+Yamux stream multiplexer implementation for py-libp2p.
+This is the preferred multiplexing protocol due to its performance and feature set.
+Mplex is also available for legacy compatibility but may be deprecated in the future.
+"""
+import logging
+import struct
+from typing import (
+    Optional,
+)
+
+import trio
+from trio import (
+    MemoryReceiveChannel,
+    MemorySendChannel,
+    Nursery,
+)
+
+from libp2p.abc import (
+    IMuxedConn,
+    IMuxedStream,
+    ISecureConn,
+)
+from libp2p.peer.id import (
+    ID,
+)
+from libp2p.stream_muxer.exceptions import (
+    MuxedStreamError,
+)
+
+PROTOCOL_ID = "/yamux/1.0.0"
+TYPE_DATA = 0x0
+TYPE_WINDOW_UPDATE = 0x1
+TYPE_PING = 0x2
+TYPE_GO_AWAY = 0x3
+FLAG_SYN = 0x1
+FLAG_ACK = 0x2
+FLAG_FIN = 0x4
+FLAG_RST = 0x8
+HEADER_SIZE = 12
+
+
+class YamuxStream(IMuxedStream):
+    def __init__(self, stream_id: int, conn: "Yamux", is_initiator: bool) -> None:
+        self.stream_id = stream_id
+        self.conn = conn
+        self.is_initiator = is_initiator
+        self.closed = False
+
+    async def write(self, data: bytes) -> None:
+        if self.closed:
+            raise MuxedStreamError("Stream is closed")
+        header = struct.pack("!BBHII", 0, TYPE_DATA, 0, self.stream_id, len(data))
```

Comment:
"!BBHII" looks like a random value here, and it's used multiple times in this file. Could you pull it up with the other constants at the top?

Reply:
Thanks for the suggestion. I’ve moved `"!BBHII"` to a constant `YAMUX_HEADER_FORMAT` at the top of `yamux.py` with a comment explaining its structure. It’s now used consistently in `write`, `close`, and `reset`.

Reply:
looks good.